internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -831,6 +831,16 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 								string(containerservice.LicenseTypeWindowsServer),
 							}, false),
 						},
+						"gmsa_dns_server": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"gmsa_root_domain_name": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
```

Comment:
Given the changes last made I'm assuming even though gmsa can be enabled without specifying a dns server and root domain it isn't a functional configuration.  In that case we can use a block, but should specify `AtLeastOneOf` so at least one or the other property must be specified for the block.

```suggestion
						"gmsa": {
							Type:     pluginsdk.TypeList,
							Optional: true,
							MaxItems: 1,
							Elem: &pluginsdk.Resource{
								Schema: map[string]*pluginsdk.Schema{
									"dns_server": {
										Type:         pluginsdk.TypeString,
										Optional:     true,
										ValidateFunc: validation.StringIsNotEmpty,
										AtLeastOneOf: []string{"windows_profile.0.gmsa.0.dns_server"},
									},
									"root_domain": {
										Type:         pluginsdk.TypeString,
										Optional:     true,
										ValidateFunc: validation.StringIsNotEmpty,
										AtLeastOneOf: []string{"windows_profile.0.gmsa.0.root_domain"},
									},
								},
							},
						},
```